holoviews/element/raster.py

```diff
@@ -517,13 +517,18 @@ class ImageStack(Image):
     - overlay/ndoverlay of Images
     """
 
+    vdims = param.List(doc="""
+        The dimension description of the data held in the matrix.""")
+
     group = param.String(default='ImageStack', constant=True)
 
     _ndim = 3
 
     _vdim_reductions = {1: Image}
 
     def __init__(self, data, kdims=None, vdims=None, **params):
+        if vdims is None:
+            raise ValueError("ImageStack requires vdims to be defined.")
```

Comment:
I felt like this was necessary because if it defaulted to only z, the other layers do not show up. On the other hand, I wasn't sure how to elegantly handle all the possible input types (dict, ds, da, tuple), so I just force the user to specify vdims.

Reply:
I think it might be nice to at least handle 3D arrays and xarray DataArray/Datasets automatically and error out for anything else.